Embedded_Code/SSA/imu/imu.ino

```diff
@@ -0,0 +1,242 @@
+           /*
+   Written and maintained by: 
+   Andrew Kettle and Jada Berenguer
+   October 26th, 2019
+*/
+
+//headers:
+
+#include "Wire.h"
+
+//Functions:
+void I2C_init();
+void getI2CData();
+int16_t convert_16bit(int8_t high, int8_t low);
+float accel_conversion(int16_t rawaccel);
+float gyro_conversion(int16_t rawgyro);
+void printData(float accelx, float accely, float accelz, float gyrox, float gyroy, float gryox);
+
+void getTemp();
+
+//pin declerations
+
+int sda = P2_2; //I2C pins
+int scl = P2_1;
+int irpin1 = A0; //3 analog IR sensors
+int irpin2 = A1;
+int irpin3 = A2;
+int sustrv = A3; //analog input for suspension travel, not implemented yet
+//int hallpin = P2.5; //Standard gpio for hall effect
+
+//read variables
+int irread1;
+int irread2;
+int irread3;
+int hallread;
+float voltage1, voltage2, voltage3;
+float temp1, temp2, temp3;
+
+//registers
+
+#define lsm9ds1_ag 0x6B //device reg
+
+#define gyro_control1 0x10 //Control of the gyro reg
+#define accel_control4 0x1E //Control of the accel reg4
+#define accel_control5 0x1F //Control of the accel reg5
+#define accel_control7 0x21 //Control of the accel reg5
+#define mag_pwr 0x16 //the register that powers the mangetometer
+//#define strt (0xD4>>1)
+#define strtw 0xD6 //send before write?
+#define strtr 0xD5
+
+#define gyroX0 0x18 //accel output registerss
+#define gyroX1 0x19
+#define gyroY0 0x1A
+#define gyroY1 0x1B
+#define gyroZ0 0x1C
+#define gyroZ1 0x1D
+
+#define accelX0 0x28 //accel output registerss
+#define accelX1 0x29
+#define accelY0 0x2A
+#define accelY1 0x2B
+#define accelZ0 0x2C
+#define accelZ1 0x2D
+
+void setup() {
+
+  pinMode(irpin1, INPUT);
+  pinMode(irpin2, INPUT);
+  pinMode(irpin3, INPUT);
+  pinMode(sustrv, INPUT);
+
+  I2C_init(); //initialize I2C transmission
+
+  Serial.begin(9600);
+}
+
+void loop() {
+
+//Note: delay is measured in milliseconds  
+
+//Reading IMU
+  getI2CData();
+  //delay(1000);
+  Serial.flush();
+}
+
+void I2C_init()
+{
+  pinMode(scl, INPUT_PULLUP);
+  pinMode(sda, INPUT_PULLUP);
```

Comment:
move to setup function

Reply:
I2C_init() is being called in setup, do you still want me to move it?